src/viam/api/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 set(PROTO_GEN_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 
-# TODO (RSDK-5364): update code to comply with up-to-date proto, then stop pinning to this commit
-set(BUF_PROTO_SOURCE buf.build/viamrobotics/api:2109757288234ffc8248121f60437cdd)
+# TODO (RSDK-5441): modify this pin through an automated process
```

Comment:
New `TODO` of [RSDK-5441](https://viam.atlassian.net/browse/RSDK-5441) here. This PR just updates our API to include https://github.com/viamrobotics/api/commit/57108c60d0da8f46df147a496845bdf2160facd6, which means the `viam::robot::v1::Status` class will now contain a `last_reconfigured` field. I investigated implementing [RSDK-4574](https://viam.atlassian.net/browse/RSDK-4573) (returning native C++ types from the robot API). Adding a native `status` type brought up a lot of questions (is `last_reconfigured` a `std::chrono::time_point`? is the inner `status` field an `AttributeMap`? how must we extend the mock robot service?) that I think I'll punt on until the C++ usability epic.